src/Sylius/Bundle/CoreBundle/EventListener/CurrentCurrencyChangeListener.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Sylius\Component\Cart\Context\CartContextInterface;
+use Sylius\Component\Cart\Context\CartNotFoundException;
+use Sylius\Component\Cart\Event\CartEvent;
+use Sylius\Component\Cart\SyliusCartEvents;
+use Sylius\Component\Core\OrderProcessing\OrderExchangeRateAndCurrencyUpdaterInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class CurrentCurrencyChangeListener
```

Comment:
I think that the name of this listener is too general. You got method ``updateCart(...)`` moreover you operates on services related with cart.